guis/common/db_classes/straw.py

```diff
@@ -145,6 +149,50 @@ def locate(self):
             .all()
         )
 
+    # same as above but returns ALL locations, not just the current one
+    # example: [<16499606623611227 straw position 84 on 1649960662278015 straw location LPAL0486 at LPALID01>, <16499477136822270 straw position 84 on 1649947713619747 straw location MN0238>]
+    def locations(self):
+        return (
+            DM.query(sl.StrawPosition)  # Get all the straw positions
+            # .join(sl.StrawPosition, sl.StrawPosition.location == sl.StrawLocation.id) # where straw positions have an entry in the straw location
+            .join(
+                sl.StrawPresent, sl.StrawPresent.position == sl.StrawPosition.id
+            )  # that also have straw present entries
+            .filter(sl.StrawPresent.straw == self.id)  # for this straw
+            .all()
+        )
+
+    # get the "present" occurance in straw_present
+    def currentPresence(self):
```

Comment:
It's querying the straw present table for this straw where this straw is marked present?

OK, but I prefer `getMyStrawPresentEntry`.

Reply:
Have a look at the analogous function in `StrawPosition` or `StrawPresent` maybe?

another try: `getStrawPresentEntry_PresentIsTrue`.